frontend/fingertips-frontend/lib/registry.tsx

```diff
@@ -0,0 +1,29 @@
+'use client';
+
+import React, { useState } from 'react';
+import { useServerInsertedHTML } from 'next/navigation';
+import { ServerStyleSheet, StyleSheetManager } from 'styled-components';
+
+export default function StyledComponentsRegistry({
+  children,
+}: Readonly<{
+  children: React.ReactNode;
+}>) {
+  // Only create stylesheet once with lazy initial state
+  // x-ref: https://reactjs.org/docs/hooks-reference.html#lazy-initial-state
```

Comment:
This doc link is to the legacy React docs. I think https://react.dev/reference/react/useState#usestate might be a better one.

Reply:
ah ok this file was a copy and paste from the nextjs recommended approach to getting styled components to render server side. Will update the link with where I found this solution.